sunkit_image/granule.py

```diff
@@ -0,0 +1,391 @@
+    elif method == 'yen':  # poor - overidentifies IG
+        threshold = skimage.filters.threshold_yen(data)
+    elif method == 'mean':  # similar to li
+        threshold = skimage.filters.threshold_mean(data)
+    elif method == 'minimum':  # terrible - identifies almost all as granule
+        threshold = skimage.filters.threshold_minimum(data)
+    elif method == 'triangle':  # overidentifies IG worse than yen
+        threshold = skimage.filters.threshold_triangle(data)
+    elif method == 'isodata':  # similar to otsu
+        threshold = skimage.filters.threshold_isodata(data)
+
+    return threshold
+
+
+def trim_intergranules(segmented_image, mark=False):
+    """
+    Remove the erronous idenfication of intergranule material in the
+    middle of granules that pure threshold segmentation produces.
+    ----------
+    Parameters:
+        segmented_image (numpy array): the segmented image containing
+                                       incorrect extra intergranules
+        mark (bool): if False, remove erronous intergranules. If True,
+                     mark them as 0.5 instead (for later examination).
+    ----------
+    Returns:
+        segmented_image_fixed (numpy array): the segmented image without
+                                             incorrect extra intergranules
+    """
+
+    if len(np.unique(segmented_image)) > 2:
+        raise ValueError('segmented_image must have only values of 1 and 0')
+
+    segmented_image_fixed = np.copy(segmented_image).astype(float)
+    labeled_seg = skimage.measure.label(segmented_image + 1, connectivity=2)
+    values = np.unique(labeled_seg)
+    # find value of the large continuous 0-valued region
+    size = 0
+    for value in values:
+        if len((labeled_seg[labeled_seg == value])) > size:
+            real_IG_value = value
+            size = len(labeled_seg[labeled_seg == value])
+
+    # set all other 0 regions to mark value (1 or 0.5)
+    for value in values:
+        if np.sum(segmented_image[labeled_seg == value]) == 0:
+            if value != real_IG_value:
+                if not mark:
+                    segmented_image_fixed[labeled_seg == value] = 1
+                elif mark:
+                    segmented_image_fixed[labeled_seg == value] = 0.5
+
+    return segmented_image_fixed
+
+
+def mark_faculae(segmented_image, data, res):
+    """
+    Mark faculae seperatley from granules - give them a value of 2 not 1.
+    ----------
+    Parameters:
+        segmented_image (numpy array): the segmented image containing
+                                incorrect middles
+        data (numpy array): the original flux values
+        res (float): Spatial resolution (arcsec/pixel) of the data
+    ----------
+    Returns:
+        segmented_image_fixed (numpy array): the segmented image with faculae
+                                             marked as 1.5
+    """
+
+    fac_size_limit = 2  # max size of a faculae in sqaure arcsec

```

Comment:
What is the reason for this being 2?

Reply:
To the best of my knowledge, 2 aquare arcsec is a resonable estimate of the maximum size that a faculae might be. I can go back and consult others on this.

Reply:
Ah so that was the reason. If possible, a citation reference here would be great.